frappe/automation/doctype/auto_repeat/auto_repeat.py

```diff
@@ -211,6 +218,68 @@ def set_auto_repeat_period(self, new_doc):
 			new_doc.set('from_date', from_date)
 			new_doc.set('to_date', to_date)
 
+	def get_next_schedule_date(self, schedule_date, for_full_schedule=False):
```

Comment:
This method seems to be very important, although I don't understand wha `schedule_date` is supposed to be set to. Mind adding a docstring to document the inputs and output values?

Reply:
Added a docstring.